sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,231 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PINMUX_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PINMUX_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+// TODO - should be autogenerated?
+#define DIF_PINMUX_MIO_NUM 32
```

Comment:
I'd file an issue and drop this comment. The name is also unclear to me, is this the number of MIOs? The count? The "number" in some other sense? It seems like it's not used very much, do DIF users need to know anything about it?

Reply:
I think you are right, I don't need to expose it to the outside world. However, before we have auto generated the enums for `dif_pinmux` in `top_earlgrey` (similar to what we have done for plic IRQs), I will have to expose `PINMUX_PARAM_NMIOPERIPHIN`, `PINMUX_PARAM_NMIOPERIPHOUT`, and couple more defines.

Reply:
This will be refactored, for now we probably still need to expose some of the defines, but eventually the corresponding enums will be generated similar to plic interrupts.

Reply:
Done.